### PR TITLE
Update voting to be on slot heights

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Votes in the system should be on slot heights, not tick heights

#### Summary of Changes
1) Change VoteProgram and associated functions/tests to be on slots rather than ticks.
2) Change replay stage to vote on slots
3) Update LeaderConfirmationService to compute confirmation on slots

Fixes #

### CODE REVIEW

File: src/leader_scheduler.rs [hunk 1]
```diff
@@ -56,9 +56,9 @@ pub struct LeaderScheduler {
     // This value must be divisible by ticks_per_slot
     pub ticks_per_epoch: u64,
 
-    // The length of time in ticks for which a vote qualifies a candidate for leader
+    // The number of slots for which a vote qualifies a candidate for leader
     // selection
-    active_window_tick_length: u64,
+    active_window_length: u64,
```
Comment (mvines) [inline src/leader_scheduler.rs]: nit: please keep the units: `active_window_slot_length`.  It's error prone otherwise, and the self-documentation is nice

Comment (carllin) [reply src/leader_scheduler.rs]: should I call it active_window_num_slots, slot length sounds like the number of ticks in a slot

Comment (garious) [reply src/leader_scheduler.rs]: Either way, can you do it in a follow-up PR? I want this one merged.

Comment (garious) [reply src/leader_scheduler.rs]: I'm doing it. I want to build off of this and I don't mind the churn of another name change.

**Pull request merged**